### PR TITLE
Doc for Canonical Aten and  Prims IR

### PR DESCRIPTION
**Pull request by SherlockNoMad**
as title. 

Sample output: https://docs-preview.pytorch.org/90644/ir.html

cc @svekars @carljparker

### CODE REVIEW

File: docs/source/ir.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+PyTorch 2.0 offers two set of IRs for backends to interface with: Canonical Aten IR and Prims IR.
+
+Canonical Aten IR
+--------------------
+
+Canonical aten ops is the core subset of aten operators that can be used to compose other operators.
+Canonical aten IR is fully functional, and there is no `inplace` or `_out` variants in this opset.
+In contrast to Prims IR, canonical aten ops reuses the existing aten ops in "native_functions.yaml",
+and it doesn't further decompose ops into explicit type promotion and boardcasting ops.
```
Comment (ezyang) [inline docs/source/ir.rst:12]: broadcasting

File: docs/source/scripts/build_opsets.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import torch._prims as prims
+
+ROOT = Path(__file__).parent.parent.parent.parent
+NATIVE_FUNCTION_YAML_PATH = ROOT / Path("aten/src/ATen/native/native_functions.yaml")
+TAGS_YAML_PATH = ROOT / Path("aten/src/ATen/native/tags.yaml")
+
+print(Path(__file__))
+print(ROOT)
+print(NATIVE_FUNCTION_YAML_PATH)
+print(TAGS_YAML_PATH)
```
Comment (ezyang) [inline docs/source/scripts/build_opsets.py:16]: logging? Probably don't want this to go to stdout?

**Comment by ezyang** [comment]
I'd love to see a sample of the generated docs

File: docs/source/scripts/build_opsets.py (hunk 2 above)
Comment (SherlockNoMad) [reply docs/source/scripts/build_opsets.py:16]: This is debugging log, will remove before check in.

Comment (SherlockNoMad) [reply docs/source/scripts/build_opsets.py:16]: These prints produces 

```
source/scripts/build_opsets.py
.
aten/src/ATen/native/native_functions.yaml
aten/src/ATen/native/tags.yaml
```

I am wondering how the doc build CI is set up? It looks like only the doc folder is present in the docker's workspace. I need to access "aten/src/ATen/native/native_functions.yaml" file to check the canonical tag. Where can I find this file in the docker env?

Comment (ezyang) [reply docs/source/scripts/build_opsets.py:16]: sorry, i don't know off the top of my head. Maybe try with-ssh/

**Comment by SherlockNoMad** [comment]
> I'd love to see a sample of the generated docs

https://docs-preview.pytorch.org/90644/ir.html

**Comment by ezyang** [comment]
This link renders as a black screen on mobile lol, I guess I'll have to check on desktop

**Review by ezyang — APPROVED** [review]
plenty more to do, but a good start

**Pull request closed without merging**